lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1337,6 +1341,23 @@ class HSADevice final : public KalmarDevice
         }
 
         return checksum.str();
+#else
+        // FNV-1a hashing, 64-bit version
+        const uint64_t FNV_prime = 0x100000001b3;
+        const uint64_t FNV_basis = 0xcbf29ce484222325;
+        uint64_t hash = FNV_basis;
+
+        const char *str = static_cast<const char *>(source);
+
+        // 104 is the proper size from Jack's research
```

Comment:
I'm still not very comfortable at merely passing the size of header to the hash algorithm. Take BrigModuleHeader as an example, originally it shall contain the hash of the BRIG module itself. But it's all filled by 0 in the current implementation. Also in LC backend we are moving toward completely ELF-compatible format so it's likely we run into same ELF headers for similar kernels.

Could you help study the impact to performance if we don't use this 104 heuristic?